src/domain/response/ResponseState.js

```diff
@@ -0,0 +1,22 @@
+const State = require('../common/State');
+
+const PROCESSED = 'Processed';
+const SENT = 'Sent';
+const FAILED = 'Failed';
+
+class ResponseState extends State {
+  constructor() {
+    super();
+    this._setState(PROCESSED, name => name === undefined);
```

Comment:
I'm not sure
```suggestion
    this._setState(PROCESSED, () => true);
```
Even though checking for `undefined` is technically true I don't think this adds much value to the business logic.